Triton_Inference_Server_Python_API/README.md

```diff
@@ -0,0 +1,265 @@
+#18 DONE 0.2s
++ [[ TEST == TRT_LLM ]]
++ [[ TEST == TEST ]]
++ mkdir -p /home/user/tutorials/Triton_Inference_Server_Python_API/models
++ cp -rf /home/user/tutorials/Triton_Inference_Server_Python_API/deps/test/test_api_models/test /home/user/tutorials/Triton_Inference_Server_Python_API/models/.
+```
+
+#### Supported Backends
+```
+dali  fil  identity  onnxruntime  openvino  python  pytorch  repeat  square  tensorflow  tensorrt
+```
+
+#### Included Models
+
+The default build includes a `test` model that can be used for
+exercising basic operations including sending input tensors of
+different data types. The `test` model copies provided inputs of
+`shape [-1, -1]` to outputs of shape `[-1, -1]`. Inputs are named
+`data_type_input` and outputs are named `data_type_output`
+(e.g. `string_input`, `string_output`).
+
+
+## Hello World
+
+### Start Container
+
+The following command starts a container and volume mounts the current
+directory as `workspace`.
+
+```bash
+   ./run.sh
+```
+
+#### Example Output
+
+```bash
+=============================
+== Triton Inference Server ==
+=============================
+
+NVIDIA Release 23.12 (build 77457706)
+Triton Server Version 2.41.0
+
+Copyright (c) 2018-2023, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+
+Various files include modifications (c) NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+
+This container image and its contents are governed by the NVIDIA Deep Learning Container License.
+By pulling and using the container, you accept the terms and conditions of this license:
+https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license
+
+NOTE: CUDA Forward Compatibility mode ENABLED.
+  Using CUDA 12.3 driver version 545.23.08 with kernel driver version 525.85.12.
+  See https://docs.nvidia.com/deploy/cuda-compatibility/ for details.
+
+root@user-machine:/workspace#
+```
+
+### Python Shell
+
+```bash
+python3
+```
+
+#### Example Output
+
+```bash
+Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
+Type "help", "copyright", "credits" or "license" for more information.
+>>>
+```
+### Create and Start a Server Instance
+
+```python
+import tritonserver
+
+server = tritonserver.Server(model_repository="/workspace/models")
+server.start()
+```
+
+### List Models
+
+```
+server.models()
+```
+
+#### Example Output
+```python
+{('test', 1): {'name': 'test', 'version': 1, 'state': 'READY'}}
+```
+
+### Send an Inference
+
+```python
+model = server.model("test")
+responses = model.infer(inputs={"string_input":[["hello world!"]]})
+```
+
+### Iterate through Responses
+
+```python
+for response in responses:
+    print(response.outputs["string_output"].to_string_array())
+```
+
+#### Example Output
+```python
+[['hello world!']]
+```
+
+
+## Stable Diffusion
+
+Please note in order to run the stable diffusion example you will need
+a hugging face token and need to set the environment variable
+`HF_TOKEN` before running the container.
```

Comment:
Perhaps mentioning that this token would be set if you run `huggingface-cli login` would be helpful.